engine/include/gen/application.hpp

```diff
@@ -4,27 +4,29 @@
 
 #include "core.hpp"
 
-#include "util/version.hpp"
+#include "gen/util/version.hpp"
 
-#include <logger/log.hpp>
-#include "graphics/device.hpp"
-#include "windowing/window.hpp"
+#include "gen/logger/log.hpp"
+#include "gen/windowing/window.hpp"
+
+#include "engine.hpp"
+#include "time.hpp"
 
 namespace gen
 {
 	class Application
 	{
 	public:
 		explicit Application(const char * appName, mim::vec2i const & initialSize);
+		virtual ~Application();
+
+		virtual void run() final;
 
-		void run();
+		virtual void draw();
 
-	private:
-		void gameLoop();
-		void shutdown();
+		virtual void update(double dt);
```

Comment:
1. Don't need `double` precision, `float` is fine.
2. Use a strong type from `chrono` instead of a unit-less type. Eg `using Seconds = std::chrono::duration<float>`.